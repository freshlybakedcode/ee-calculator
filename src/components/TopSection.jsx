import styled from 'styled-components';

const SolarContainer = styled.div`
  background: linear-gradient(
    135deg,
    hsla(0, 0%, 10%, 0.3) 0%,
    hsla(263, 16%, 22%, 0.2) 100%
  );
  border-bottom-left-radius: 10px;
  border-bottom-right-radius: 10px;
  box-shadow: 2px 3px 4px 1px hsla(0, 0%, 10%, 0.3);
  display: flex;
  justify-content: flex-end;
  margin-bottom: 15px;
  padding: 25px 25px 20px 25px;
`;

const SolarPanelHolder = styled.div`
  border-radius: 5px;
  border: 3px solid #111;
  display: inline-flex;
  overflow: hidden;
`;

const SolarPanel = styled.div`
  background: linear-gradient(#524243 0%, #533d34 100%);
  border-right: solid 1px #776561;
  height: 20px;
  width: 20px;

  &:last-child {
    border-right: none;
  }
`;

export const TopSection = () => (
  <SolarContainer>
    <SolarPanelHolder>
      <SolarPanel />
      <SolarPanel />
      <SolarPanel />
      <SolarPanel />
    </SolarPanelHolder>
  </SolarContainer>
);
