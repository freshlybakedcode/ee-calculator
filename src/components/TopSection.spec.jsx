import { render } from '@testing-library/react';
import { TopSection } from './TopSection';

describe('TopSection', () => {
  it('should render as expected', () => {
    const { container } = render(<TopSection />);
    expect(container).toMatchSnapshot();
  });
});
