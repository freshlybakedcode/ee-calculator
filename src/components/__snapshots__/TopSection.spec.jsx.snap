// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopSection should render as expected 1`] = `
<div>
  <div
    class="sc-bdnxRM cxLqHU"
  >
    <div
      class="sc-gtsrHT lkAkzf"
    >
      <div
        class="sc-dlnjwi hhIhYm"
      />
      <div
        class="sc-dlnjwi hhIhYm"
      />
      <div
        class="sc-dlnjwi hhIhYm"
      />
      <div
        class="sc-dlnjwi hhIhYm"
      />
    </div>
  </div>
</div>
`;
